src/daemon/daemon.js

```diff
@@ -168,9 +168,53 @@ const parseCfgMultiaddr = (addr) => (addr.includes('/http')
   : multiaddr(addr).encapsulate('/http')
 )
 
+async function checkPortsArray (addrs) {
+  for (const addr of addrs) {
+    const ma = parseCfgMultiaddr(addr)
+    const port = parseInt(ma.nodeAddress().port, 10)
+    const isDaemon = await checkIfAddrIsDaemon(ma.nodeAddress())
+
+    if (isDaemon) {
+      continue
+    }
+
+    const freePort = await getPort({ port: getPort.makeRange(port, port + 100) })
+
+    if (port !== freePort) {
+      showDialog({
+        title: i18n.t('multipleBusyPortsDialog.title'),
+        message: i18n.t('multipleBusyPortsDialog.message'),
+        type: 'error',
+        buttons: [
+          i18n.t('multipleBusyPortsDialog.action')
+        ]
+      })
+
+      throw new Error('ports already being used')
+    }
+  }
+}
+
 async function checkPorts (ipfsd) {
   const config = readConfigFile(ipfsd)
 
+  const apiIsArr = Array.isArray(config.Addresses.API)
+  const gatewayIsArr = Array.isArray(config.Addresses.Gateway)
+
+  if (apiIsArr || gatewayIsArr) {
+    logger.info('[daemon] custom configuration with array of API or Gateway addrs')
+
+    let addrs = apiIsArr
+      ? config.Addresses.API
+      : [config.Addresses.API]
+
+    addrs = gatewayIsArr
+      ? addrs.concat(config.Addresses.Gateway)
+      : addrs.concat([config.Addresses.Gateway])
+
+    return checkPortsArray(addrs)
```

Comment:
Up to you if you want to change but you can simplify this loads to just:

```suggestion
    return checkPortsArray([].concat(config.Addresses.API, config.Addresses.Gateway))
```

You can pass arrays to `concat` and it'll concat all the items so it doesn't matter if the config is an array or not:

```js
> const foo = [1, 2, 3]
undefined
> foo.concat([4, 5, 6], 7, [8, 9, 10])
[
  1, 2, 3, 4, 5,
  6, 7, 8, 9, 10
]
```